components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java

```diff
@@ -241,18 +241,22 @@ private boolean isPreserverClaimUrisInAssertion(OAuthTokenReqMessageContext requ
     }
 
     private Map<ClaimMapping, String> getCachedUserAttributes(OAuthTokenReqMessageContext requestMsgCtx) {
-        Map<ClaimMapping, String> userAttributes = getUserAttributesCachedAgainstToken(getAccessToken(requestMsgCtx));
+
+        Map<ClaimMapping, String> userAttributes =
+                getUserAttributesCachedAgainstAuthorizationCode(getAuthorizationCode(requestMsgCtx));
         if (log.isDebugEnabled()) {
-            log.debug("Retrieving claims cached against access_token for user: " + requestMsgCtx.getAuthorizedUser());
+            log.debug("Retrieving claims cached against authorization_code for user: " +
+                    requestMsgCtx.getAuthorizedUser());
         }
         if (isEmpty(userAttributes)) {
             if (log.isDebugEnabled()) {
-                log.debug("No claims cached against the access_token for user: " + requestMsgCtx.getAuthorizedUser() +
-                        ". Retrieving claims cached against the authorization code.");
+                log.debug("No claims cached against the authorization_code for user: " +
+                        requestMsgCtx.getAuthorizedUser() +
+                        ". Retrieving claims cached against the access_token code.");
             }
-            userAttributes = getUserAttributesCachedAgainstAuthorizationCode(getAuthorizationCode(requestMsgCtx));
+            userAttributes = getUserAttributesCachedAgainstToken(getAccessToken(requestMsgCtx));
```

Comment:
We have to give priority to access token..
if it presents get the attribute from access token...

else check authorization code presents and if exits get user attributes from authorization code.

Reply:
+1

At the point of issuing an id_token in ideal case the attributes stored against code and token should be same. Since there is a step to copy attributes cached against code to token in the authorization code flow.

If the attributes cached against the token are not updated then the issue might be else where

Reply:
yes. The issue was due to copying the attributes cached against code to token in the authorization code flow before building the ID token. Fixed it by moving the `addUserAttributesAgainstAccessToken` method before the ID token build.